packages/pluggableWidgets/barcode-scanner-web/src/components/BarcodeScanner.tsx

```diff
@@ -64,35 +45,26 @@ export function BarcodeScannerOverlay({
     );
 }
 
+export interface BarcodeScannerProps extends Dimensions {
+    onDetect?: (data: string) => void;
+    showMask: boolean;
+    class: string;
+}
+
 export function BarcodeScanner({
     onDetect,
     showMask,
     class: className,
     ...dimensions
 }: BarcodeScannerProps): ReactElement | null {
-    const videoElement = useRef<HTMLVideoElement | null>(null);
-    const hasDetectedOnce = useRef<boolean>(false);
-    const { streamObject, error: errorMediaStream } = useMediaStream();
-    const { codeResult, error: errorCodeScanner } = useCodeScanner(streamObject, videoElement.current);
-    const supportsCameraAccess = browserSupportsCameraAccess();
-
-    // If both the video element ref and the camera stream object are ready, display it through the `srcObject` prop.
-    useEffect(() => {
-        if (videoElement.current && streamObject) {
-            videoElement.current.srcObject = streamObject;
+    const [errorMessage, setError] = useCustomErrorMessage();
+    const videoRef = useReader({ onSuccess: onDetect, onError: setError });
+    const supportsCameraAccess = useHasUserMedia();
```

Comment:
This on seems unnecessary. Just one check, without any hook logic.
I believe we can remove the file and method and simply put 
`const supportsCameraAccess = typeof navigator?.mediaDevices?.getUserMedia === "function";`

Reply:
it's better to use hook as in unit tests we can mock hook behaviour

Reply:
Why we need to mock hook behaviour if it's not a hook?

Reply:
it's just simpler then mocking whole `naviagor.mediaDevices` api